### DISCUSSION TITLE
Implicit Usings

### DISCUSSION THREAD

**Discussion by liamlim** [dotnet/runtime#61157] [Ideas] [answered]
Hello, I have some feedback for global usings and implicit usings

1) Do not use IDE0005 for unused global usings. 
Reason: I want to be warned about unused using which is usually created by refactoring and forgetting to check usings header. In contrary for global usings I specifically ask for importing namespace into the scope. I explicitly specify in one place that I want to import a specific using into the scope. IDE0005 is redundant.

2) Please provide an easy way to disable implicit usings for new project
Reason: It's annoying to modify 'enable' to 'disable' for every single project that I create and add into solution. I want to specify in Visual Studio settings that ALL projects which I create should have implicit usings disabled. Ideally provide a way to specify it in .editorconfig file so it applies also for projects created by other developers working on our project

3) Please let us specify in Visual Studio that every newly created project has a file Global.cs and add all implicit usings there. Then it's clear which usings are actually imported into the scope and user can add any global using they want (or remove). It's much better solution that import implicit usings during build which makes no sense to me.

Honestly I do not like the implicit global usings feature at all. It's not clear which usings are actually imported into the scope for which project. Also adding another implicit using by C# team will be a breaking change because implicit usings are added into the scope during build and not project creation. This is just wrong imo.

**Comment by CyrusNajmabadi** [comment, marked as answer]
Moving to sdk team. Implicit usings aren't part of Roslyn.
